app/src/main/java/com/hedvig/app/feature/home/ui/changeaddress/GetSelfChangeEligibilityUseCase.kt

```diff
@@ -0,0 +1,35 @@
+package com.hedvig.app.feature.home.ui.changeaddress
+
+import com.apollographql.apollo.ApolloClient
+import com.hedvig.android.owldroid.graphql.SelfChangeEligibilityQuery
+import com.hedvig.android.owldroid.type.SelfChangeBlocker
+import com.hedvig.app.feature.home.ui.changeaddress.GetSelfChangeEligibilityUseCase.SelfChangeEligibilityResult.Blocked
+import com.hedvig.app.feature.home.ui.changeaddress.GetSelfChangeEligibilityUseCase.SelfChangeEligibilityResult.Eligible
+import com.hedvig.app.feature.home.ui.changeaddress.GetSelfChangeEligibilityUseCase.SelfChangeEligibilityResult.Error
+import com.hedvig.app.util.apollo.QueryResult
+import com.hedvig.app.util.apollo.safeQuery
+
+class GetSelfChangeEligibilityUseCase(
```

Comment:
How does a `UseCase` differ from a `Repository`? Is the difference merely in name? 🤔

Reply:
Or is it that it crunches the data from the queries and returns what the `Activity`/`Fragment` previously used to do? 😄

Reply:
I think [this](https://stackoverflow.com/questions/43055247/what-is-the-difference-between-repositories-and-usecases) SO thread explains it quite nice: 

"Simply, Use-Cases handles your business logic, Repositories are the data layer which you store and access data."

For example, `ChangeAddressViewModel` calls use cases in order to construct a `ViewState`, we could skip the use cases and do this in the view model as well but I think use cases encapsulate business logic quite well. If we had local storage (eg. room), we could have a `MemberRepository` which would handle network and database requests. The use case would then call `memberRepository.selfChangeEligibility()` instead of `apolloClient.query(SelfChangeEligibilityQuery()).safeQuery()`. 

Hope it makes sense!

Reply:
It does indeed! 🙇